plugins/c9.vfs.client/vfs_client.js

```diff
@@ -193,12 +193,12 @@ define(function(require, exports, module) {
                 extraPaths = path;
                 path = path[0];
                 extraPaths = "," + extraPaths.map(function(p) {
-                    return p[0] == path[0] && p != path ? escape(p) : "";
+                    return p[0] == path[0] && p != path ? encodeURI(p) : "";
```

Comment:
@nightwing I think the extra paths should explicitly escape commas in filenames, because they are part of the convention of the download API, e.g.:

                extraPaths = "," + extraPaths.map(function(p) {
                    return p[0] == path[0] && p != path ? encodeURI(p).replace(/,/g, '%2C') : "";
                }).filter(Boolean).join(",");

Using `encodeURI` instead of `escape` ensures that other filename characters such as `+` will be correctly URI-encoded.